api4/post.go

```diff
@@ -334,11 +347,19 @@ func updatePost(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if !c.App.SessionHasPermissionToPost(c.Session, c.Params.PostId, model.PERMISSION_EDIT_OTHERS_POSTS) {
-		c.SetPermissionError(model.PERMISSION_EDIT_OTHERS_POSTS)
+	originalPost, err := c.App.GetSinglePost(c.Params.PostId)
+	if err != nil {
+		c.SetPermissionError(model.PERMISSION_EDIT_POST)
```

Comment:
Why return a permission error here? Won't the error be unrelated to permissions if one does occur? Or are you assuming if it errors it means the post doesn't exist, so we want to respond with a permissions error in that case

Reply:
If it's the same behaviour as before, that's fine with me